### ISSUE TITLE
Failed to install IKOS v3.1 with Homebrew on Ubuntu 20.04 (WSL-2)

### ISSUE THREAD

**Issue by ThibFrgsGmz** [NASA-SW-VnV/homebrew-core#5]
I tried to install [IKOS v3.1](https://github.com/NASA-SW-VnV/ikos/releases/tag/v3.1) with Homebrew as described in the [project installation procedure](https://github.com/NASA-SW-VnV/ikos#installation): 

```
brew install nasa-sw-vnv/core/ikos
```

However I end up with errors:

```sh
...
==> Installing nasa-sw-vnv/core/ikos dependency: apron
==> Patching
==> make APRON_PREFIX=/home/linuxbrew/.linuxbrew/Cellar/apron/0.9.10_1 GMP_PREFIX=/home/linuxbrew/.linuxbrew/opt/gmp MPFR_PREFIX=/home/linuxbrew/.li
Last 15 lines from /home/ThibFrgsGmz/.cache/Homebrew/Logs/apron/01.make:

(cd num; make all)
make[1]: Entering directory '/tmp/apron-20221219-27408-1nwfqzz/apron-0.9.10/num'
make[1]: Nothing to be done for 'all'.
make[1]: Leaving directory '/tmp/apron-20221219-27408-1nwfqzz/apron-0.9.10/num'
(cd itv; make all)
make[1]: Entering directory '/tmp/apron-20221219-27408-1nwfqzz/apron-0.9.10/itv'
gcc-10 -Wcast-qual -Wswitch -Werror-implicit-function-declaration -Wall -Wextra -Wundef -Wbad-function-cast -Wcast-align -Wstrict-prototypes -Wno-unused -std=c99 -U__STRICT_ANSI__ -fPIC -O3 -DNDEBUG -I../num -I../apron -I/home/linuxbrew/.linuxbrew/opt/gmp/include -I/home/linuxbrew/.linuxbrew/opt/mpfr/include -DNUM_MPQ -c -o itvMPQ.o itv.c
gcc-10: internal compiler error: Segmentation fault signal terminated program as
Please submit a full bug report,
with preprocessed source if appropriate.
See <file:///usr/share/doc/gcc-10/README.Bugs> for instructions.
make[1]: *** [Makefile:146: itvMPQ.o] Error 4
make[1]: Leaving directory '/tmp/apron-20221219-27408-1nwfqzz/apron-0.9.10/itv'
make: *** [Makefile:20: c] Error 2

If reporting this issue please do so at (not Homebrew/brew or Homebrew/core):
  https://github.com/nasa-sw-vnv/homebrew-core/issues
```

Here are the complete contents of the `/home/ThibFrgsGmz/.cache/Homebrew/Logs/apron/01.make' file.`

```sh
2022-12-19 14:13:58 +0000

make
APRON_PREFIX=/home/linuxbrew/.linuxbrew/Cellar/apron/0.9.10_1
GMP_PREFIX=/home/linuxbrew/.linuxbrew/opt/gmp
MPFR_PREFIX=/home/linuxbrew/.linuxbrew/opt/mpfr
PPL_PREFIX=/home/linuxbrew/.linuxbrew/opt/ppl
HAS_OCAML=
HAS_OCAMLOPT=
HAS_JAVA=
OCAMLFIND=
HAS_PPL=1
all
install

(cd num; make all)
make[1]: Entering directory '/tmp/apron-20221219-27408-1nwfqzz/apron-0.9.10/num'
make[1]: Nothing to be done for 'all'.
make[1]: Leaving directory '/tmp/apron-20221219-27408-1nwfqzz/apron-0.9.10/num'
(cd itv; make all)
make[1]: Entering directory '/tmp/apron-20221219-27408-1nwfqzz/apron-0.9.10/itv'
gcc-10 -Wcast-qual -Wswitch -Werror-implicit-function-declaration -Wall -Wextra -Wundef -Wbad-function-cast -Wcast-align -Wstrict-prototypes -Wno-unused -std=c99 -U__STRICT_ANSI__ -fPIC -O3 -DNDEBUG -I../num -I../apron -I/home/linuxbrew/.linuxbrew/opt/gmp/include -I/home/linuxbrew/.linuxbrew/opt/mpfr/include -DNUM_MPQ -c -o itvMPQ.o itv.c
gcc-10: internal compiler error: Segmentation fault signal terminated program as
Please submit a full bug report,
with preprocessed source if appropriate.
See <file:///usr/share/doc/gcc-10/README.Bugs> for instructions.
make[1]: *** [Makefile:146: itvMPQ.o] Error 4
make[1]: Leaving directory '/tmp/apron-20221219-27408-1nwfqzz/apron-0.9.10/itv'
make: *** [Makefile:20: c] Error 2

HOMEBREW_VERSION: 3.6.15
ORIGIN: https://github.com/Homebrew/brew
HEAD: 11cdffb4fef84828a7e02860212b30ce4fac85f5
Last commit: 10 days ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: a77e76758f6e1e1aafdf28adf2849f1996c19311
Core tap last commit: 10 minutes ago
Core tap branch: master
HOMEBREW_PREFIX: /home/linuxbrew/.linuxbrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_MAKE_JOBS: 3
HOMEBREW_NO_BOOTSNAP: set
Homebrew Ruby: 2.6.8 => /home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.8_1/bin/ruby
CPU: 3-core 64-bit skylake
Clang: 10.0.0
Git: 2.25.1 => /bin/git
Curl: 7.68.0 => /bin/curl
Kernel: Linux 4.19.128-microsoft-standard x86_64 GNU/Linux
OS: Ubuntu 20.04.5 LTS (focal)
Host glibc: 2.31
/usr/bin/gcc: 10.3.0
/usr/bin/ruby: N/A
glibc: 2.35_1
gcc@11: N/A
gcc: 12.2.0
xorg: N/A

HOMEBREW_CC: gcc-10
HOMEBREW_CXX: g++-10
CMAKE_PREFIX_PATH: /home/linuxbrew/.linuxbrew/opt/glibc:/home/linuxbrew/.linuxbrew/opt/binutils:/home/linuxbrew/.linuxbrew
PKG_CONFIG_PATH: /home/linuxbrew/.linuxbrew/opt/gmp/lib/pkgconfig:/home/linuxbrew/.linuxbrew/opt/mpfr/lib/pkgconfig:/home/linuxbrew/.linuxbrew/opt/isl/lib/pkgconfig:/home/linuxbrew/.linuxbrew/opt/lz4/lib/pkgconfig:/home/linuxbrew/.linuxbrew/opt/xz/lib/pkgconfig:/home/linuxbrew/.linuxbrew/opt/zlib/lib/pkgconfig:/home/linuxbrew/.linuxbrew/opt/zstd/lib/pkgconfig
HOMEBREW_GIT: git
ACLOCAL_PATH: /home/linuxbrew/.linuxbrew/share/aclocal
PATH: /home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/shims/linux/super:/home/linuxbrew/.linuxbrew/opt/glibc/bin:/home/linuxbrew/.linuxbrew/opt/lz4/bin:/home/linuxbrew/.linuxbrew/opt/xz/bin:/home/linuxbrew/.linuxbrew/opt/zstd/bin:/home/linuxbrew/.linuxbrew/opt/binutils/bin:/home/linuxbrew/.linuxbrew/opt/gcc/bin:/home/linuxbrew/.linuxbrew/opt/ppl/bin:/usr/bin:/bin:/usr/sbin:/sbin

```

**Comment by ThibFrgsGmz** [comment]
FYI: I uninstalled and reinstalled Homebrew (with this [tutorial](https://www.how2shout.com/linux/install-brew-on-wsl-windows-subsystem-for-linux/)) and the problem is still there.

**Comment by disconnect3d** [comment]
I also hit this issue when installing on MacOS.

**Comment by darked89** [comment]
I got:

```
==> make install
Last 15 lines from /home/darked89/.cache/Homebrew/Logs/ikos/04.make:
make[2]: *** Waiting for unfinished jobs....
[ 11%] Building CXX object frontend/llvm/CMakeFiles/ikos-pp-lib.dir/src/pass/name_values.cpp.o
/tmp/ikos-20230417-23905-11is14a/ikos-3.1/ar/src/pass/add_partitioning_variables.cpp:59:50: error: variable 'constexpr const std::array<unsigned int, 4> ikos::ar::bit_widths' has initializer but incomplete type
   59 | constexpr static const std::array< unsigned, 4 > bit_widths = {1, 8, 32, 64};
      |                                                  ^~~~~~~~~~
make[2]: *** [ar/CMakeFiles/ikos-ar.dir/build.make:160: ar/CMakeFiles/ikos-ar.dir/src/pass/pass.cpp.o] Error 1
[ 12%] Building CXX object frontend/llvm/CMakeFiles/ikos-pp-lib.dir/src/pass/remove_printf_calls.cpp.o
[ 12%] Building CXX object frontend/llvm/CMakeFiles/ikos-pp-lib.dir/src/pass/remove_unreachable_blocks.cpp.o
make[2]: *** [ar/CMakeFiles/ikos-ar.dir/build.make:132: ar/CMakeFiles/ikos-ar.dir/src/pass/add_partitioning_variables.cpp.o] Error 1
make[2]: *** [ar/CMakeFiles/ikos-ar.dir/build.make:188: ar/CMakeFiles/ikos-ar.dir/src/pass/simplify_upcast_comparison.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:1497: ar/CMakeFiles/ikos-ar.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
[ 14%] Linking CXX static library libikos-pp.a
[ 14%] Built target ikos-pp-lib
make: *** [Makefile:146: all] Error 2
```
WSL-2, Ubuntu 20.04, gcc (Debian 10.2.1-6), Debian clang version 11.0.1-2

**Comment by mschwerhoff** [comment]
Similar situation (seems related to issues #5 and #6):

```
$ wsl.exe -l -v
  NAME                   STATE           VERSION
* Ubuntu-20.04           Running         2
  ...
```

```
$ uname -a
Linux streamer 5.15.90.1-microsoft-standard-WSL2 #1 SMP Fri Jan 27 02:56:13 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux`
```

```
$ brew install nasa-sw-vnv/core/ikos
...
==> Installing nasa-sw-vnv/core/ikos
==> python3 -m venv --system-site-packages /home/linuxbrew/.linuxbrew/Cellar/ikos/3.1_1/libexec/vendor
==> /home/linuxbrew/.linuxbrew/Cellar/ikos/3.1_1/libexec/vendor/bin/pip install -v --no-deps --no-binary :all: --use-feature=no-binary-enable-wheel-cache --i
==> cmake -G Unix Makefiles -DGMP_ROOT=/home/linuxbrew/.linuxbrew/opt/gmp -DMPFR_ROOT=/home/linuxbrew/.linuxbrew/opt/mpfr -DPPL_ROOT=/home/linuxbrew/.linuxbr
==> make install
Last 15 lines from /home/malte/.cache/Homebrew/Logs/ikos/04.make:
   51 | #include <ikos/ar/support/number.hpp>
  +++ |+#include <vector>
   52 |
/tmp/ikos-20230530-10071-etz05b/ikos-3.1/ar/include/ikos/ar/semantic/data_layout.hpp:114:8: error: ‘vector’ in namespace ‘std’ does not name a template type
  114 |   std::vector< DataLayoutInfo > floats;
      |        ^~~~~~
/tmp/ikos-20230530-10071-etz05b/ikos-3.1/ar/include/ikos/ar/semantic/data_layout.hpp:114:3: note: ‘std::vector’ is defined in header ‘<vector>’; did you forget to ‘#include <vector>’?
  114 |   std::vector< DataLayoutInfo > floats;
      |   ^~~
make[2]: *** [ar/CMakeFiles/ikos-ar.dir/build.make:160: ar/CMakeFiles/ikos-ar.dir/src/pass/pass.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:1497: ar/CMakeFiles/ikos-ar.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
[ 12%] Linking CXX static library libikos-pp.a
[ 12%] Built target ikos-pp-lib
make: *** [Makefile:146: all] Error 2

If reporting this issue please do so at (not Homebrew/brew or Homebrew/homebrew-core):
  https://github.com/nasa-sw-vnv/homebrew-core/issues

These open issues may also help:
IKOS build fails on gcc12 https://github.com/NASA-SW-VnV/homebrew-core/issues/6
Failed to install IKOS v3.1 with Homebrew on Ubuntu 20.04 (WSL-2) https://github.com/NASA-SW-VnV/homebrew-core/issues/5
```

**Comment by ivanperez-keera** [comment]
@mschwerhoff This is related to https://github.com/NASA-SW-VnV/ikos/issues/225. This bug has been fixed (see: https://github.com/NASA-SW-VnV/ikos/issues/225#issuecomment-1763392141) but we have not published the release, or put it in homebrew yet.

We plan to put out a new release of ikos soon and this bug should go away.

**Comment by ivanperez-keera** [comment]
@darked89 Are you able to try this PR: https://github.com/NASA-SW-VnV/ikos/pull/234

Let me know if it works (or doesn't). Thanks!

**Comment by ivanperez-keera** [comment]
Hi everyone! We've just announced a release candidate for IKOS 3.2, and we've included instructions to install it with homebrew. See https://github.com/NASA-SW-VnV/ikos/discussions/253 for details. Let me know if this issue persists. Thanks!

**Comment by ivanperez-keera** [comment]
Closed by the solution to #9.
